flytekit/taskplugins/sqlite/task.py

```diff
@@ -0,0 +1,113 @@
+import contextlib
+import os
+import sqlite3
+import tempfile
+import typing
+from dataclasses import dataclass
+from enum import Enum
+
+from flytekit import FlyteContext, kwtypes
+from flytekit import metadata as md
+from flytekit.annotated.base_sql_task import SQLTask
+from flytekit.models import task as _task_models
+from flytekit.plugins import pandas as pd
+from flytekit.types import FlyteFile, FlyteSchema
+from flytekit.types.flyte_file import unarchive_file
+
+
+@dataclass
+class SQLite3Config(object):
+    """
+    Use this configuration to configure if sqlite3 files are passed in dynamically into the task or
+    to configure a path to a static sqlite3 compatible database file that is available either
+      - within the container
+      - or from a publicly downloadable source
+
+    If the Mode is marked dynamic, then a variable called ``sqlite: FlyteFile[".sqlite"]`` will be automatically added
+    to the input interface of the task. Failing to supply this at runtime will raise an Assertion
+
+        Args:
+            sqlite_db_mode: STATIC should be use for constant locations of file, dynamic to allow dynamically provided
+                            sqlite3 file locations. The file will have to comply with FlyteFile[.db|.zip|.tar.gz|.bz2]
+                            format
+            static_file_path: default FlyteFile that will be downloaded on execute
+            compressed: Boolean that indicates if the given file is a compressed archive
```

Comment:
worth including which compressed file types are supported? or maybe just link to shutil docs?

Reply:
+1